src/main/java/io/nats/client/api/ConsumerConfiguration.java

```diff
@@ -298,17 +298,102 @@ public long getMaxPullWaiting() {
      * Get the header only flag indicating whether it's on or off
      * @return the flow control mode
      */
-    public boolean getHeadersOnly() {
+    public boolean isHeadersOnly() {
         return headersOnly != null && headersOnly;
     }
 
+    /**
+     * Gets whether deliver policy of this consumer configuration was set or left unset
+     * @return true if the policy was set, false if the policy was not set
+     */
+    public boolean deliverPolicyWasSet() {
```

Comment:
Not objecting per se, but do all of these need to be public?  Anything public must be supported until at least the next major version.

Reply:
It's a packaging problem. Not sure how to get around it all.